editors/code/src/main.ts

```diff
@@ -188,6 +189,46 @@ async function bootstrapServer(config: Config, state: PersistentState): Promise<
     return path;
 }
 
+async function patchelf(dest: PathLike): Promise<void> {
+    await vscode.window.withProgress(
+        {
+            location: vscode.ProgressLocation.Notification,
+            title: "Patching rust-analyzer for NixOS"
+        },
+        async (progress, _) => {
+            const patchPath = path.join(os.tmpdir(), "patch-ra.nix");
+            progress.report({ message: "Writing nix file", increment: 5 });
+            await fs.writeFile(patchPath, `
+            {src, pkgs ? import <nixpkgs> {}}:
+                pkgs.stdenv.mkDerivation {
+                    name = "rust-analyzer";
+                    inherit src;
+                    phases = [ "installPhase" "fixupPhase" ];
+                    installPhase = "cp $src $out";
+                    fixupPhase = ''
+                    chmod 755 $out
+                    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $out
+                    '';
+                }
+            `);
```

Comment:
Is there perhaps some way to specify the scritp on the command line, without writing an inermediate file?

Reply:
Yes there is, but I don't know how I can find path to it in installed extension :(

Reply:
I'll try making it not ugly)

Reply:
This should be alright